### DISCUSSION TITLE
How do i capture error message?

### DISCUSSION THREAD

**Discussion by TinyTimothy2** [TabularEditor/TabularEditor3#1232] [🙋♀️TE3 Q&A] [answered]
How do i capture error message in TE 3
Ie  I have a semantic error and i highlite the underscore and i can read the message in a pop up  I need to capture it so i can learn more about it.
In Visual Studio you can copy the error message

**Comment by otykier** [comment, marked as answer]
Hi @TinyTimothy2 
If you navigate to the **Messages** view, you can select and copy (Ctrl+C) the error message:

<img width="788" alt="image" src="https://github.com/TabularEditor/TabularEditor3/assets/8976200/95a69c96-b963-4bab-91f7-59133f262c6d">

For example, if I copy the highlighted row, I get the following into the clipboard:

| Object | Type | Message | Line | Column |
| --- | --- | --- | --- | --- |
| | Warning | The model already contains a measure named [Sales]. Using this name for an extended column may cause ambiguity. | 2 | 34 |

**Comment by Eivind4** [comment]
Would it be possible to easily get only the error message? Such as right-click and copy message?
Did not know about Ctrl+C option before I researched this post. 

Copying error messages is very handy when using LLMs when developing and you need to debug scripts assisted with LLMs.

Assume Visual Studio is an alternative as commented in the original post.

**Reply by otykier** [reply]
We don't have such an option today, but that's a good suggestion, so I've added it to our backlog.

**Comment by otykier** [comment]
Hi all.
Glad to inform you that we now have improved this behavior in [3.23.0](https://docs.tabulareditor.com/te3/other/release-notes/3_23_0.html), so that Ctrl+C will copy just the value in the focused cell, while Ctrl+Shift+C will copy the full row. There's also a right-click context menu where you can select these options:
<img width="1479" height="296" alt="image" src="https://github.com/user-attachments/assets/01b7c27c-80d6-4c72-8019-d93ae2245ec3" />

**Comment by Eivind4** [comment]
Great news!
